core/selection.js

```diff
@@ -2018,6 +2018,8 @@
 
 			// Handle special case - fake selection of table cells.
 			if ( editor && editor.plugins.tableselection &&
+				// (#2945)
+				!editor.plugins.tableselection.isElementIgnored( ranges[ 0 ] && ranges[ 0 ].startContainer ) &&
```

Comment:
Shouldn't it be checked after all other checks? Additionally it would be probably better to invoke it on `range._getTableElement`, as whole table selection can be hooked directly in editable, so `ranges[ 0 ].startContainer` would be pointing to `body`/`div`.